test/unit/loops.spec.ts

```diff
@@ -621,3 +601,28 @@ test("for...in with pre-defined variable keeps last value", () => {
         return x;
     `.expectToMatchJsResult();
 });
+
+test.each([
+    ["foo", "['bar']"],
+    ["[foo]", "[['bar']]"],
+    ["{ foo }", "[{ foo: 'bar' }]"],
+])("forof assignment updates dependencies", (initializer, expression) => {
+    util.testModule`
+        let foo = '';
+        export { foo };
+        for (${initializer} of ${expression}) {}
+    `
+        .setReturnExport("foo")
+        .expectToEqual("bar");
+});
+
+test.each([
+    ["const [foo]", "[['bar']]"],
+    ["const { foo }", "[{ foo: 'bar' }]"],
+])("forof variable declaration binding patterns", (initializer, expression) => {
```

Comment:
Can we run it on all cases in `destructuring.spec.ts` now?

Reply:
Could run them on the tests you specified but couldn't expect the result to match to JS result on the other since TypeScript has a bug where reassigning a value that is exported is not actually updated.

[Babel's Transpilation](https://babeljs.io/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=DYUwLgBAZg9jEF4IHJkG4BQIAeAHGATpAN7RwQC-msBEAFLPDFBANoBEAbgIbACuIdgF0AlBGIUgA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015&prettier=false&targets=&version=7.8.4&externalPlugins=)

[TypeScript's Transpilation](https://www.typescriptlang.org/play/?target=2&module=1#code/DYUwLgBAZg9jEF4IHJkG4CwAoEAPADjAE6QDe0cEAvplrERABSzwxQQDaARAG4CGwAK4guAXQCUEclWxA)